src/Measurements/eccentricity.jl

```diff
@@ -0,0 +1,313 @@
+"""
+    struct Threaded <: AbstractGraphAlgorithm
+
+A struct representing a threaded version of a measurement algorithm.
+"""
+struct Threaded end
+
+"""
+    struct Eccentricities <: GraphMeasurement
+
+A structure representing the eccentricities calculated from a given graph.
+"""
+struct Eccentricities{T<:Real} <: GraphMeasurement
+    vals::Vector{T}
+    min::T
+    max::T
+end
+Eccentricities(a::AbstractVector{<:Real}) = Eccentricities(a, extrema(a)...)
+
+function _eccentricities(g::AbstractGraph,
+                         vs::AbstractVector{<:Integer},
+                         distmx::AbstractMatrix{T},
+                         alg::SSSPAlgorithm,
+                         use_vs::Bool,
+                         use_dists::Bool,
+                         ET) where {T<:Real}
+
+    verts = use_vs ? vs : vertices(g)
+
+    # BFS and Dijkstra across vertices are faster than APSP for sparse graphs.
+
+    eccs = Vector{ET}()
+    sizehint!(eccs, length(verts))
+    mn = typemax(ET)
+    mx = typemin(ET)
+    if use_dists
+        @inbounds for v in verts
```

Comment:
```suggestion
        for v in verts
```
there is no bound checking in this loop